File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n=len(nums)

        #0123456 -> 5601234

        

        def poi(l, r):
            if l>=r:
                return l
            mid=(l+r)//2
            if nums[mid]<nums[mid-1] and nums[mid]<nums[mid+1]:
                return mid
            elif nums[mid]>nums[r]:
                return poi(mid+1,r)
            elif nums[mid]<nums[r]:
                return poi(l,mid)

        l,r=0,n-1
        return nums[poi(l,r)]

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> diff = new HashMap<>();
        int[] indices = new int[2];
        for(int i=0; i<nums.length; i++) {
            diff.put(nums[i], i);
        }

        for(int j=0; j<nums.length; j++) {
            if(diff.containsKey(target-nums[j]) && j!=diff.get(target-nums[j])) {
                indices[0] = j;
                indices[1] = diff.get(target-nums[j]);
                return indices;
            }
        }
        return indices;
    }
}
